### PR TITLE
feat(trading): activity streaks, reward hoarder bonus and active rewards

### LINKED ISSUE #4532: Update rewards view
## Story

As a user 
I want to see details of various discounts / rewards I'm getting 
So that I feel like a valued customer and can tell what I should be doing to earn / maximise my rewards

## Acceptance Criteria

- [x] I can see the sketches in the figma
- [x] For now we will put this in the top level nav (until we can rework portfolio properly)
- [ ] All the numbers work
- [ ] There is a disconnected state that is still useful
  - [ ] The tiers are still shown just with no progress
  - [ ] The reward pool would be empty
  - [x] Vesting base rate would still show
  - [x] The multpliers would be blank 
  - [x] The reward history should be shown for the network rather than for the user 
- [x] There is a way to toggle the reward history to show "Earned by me" vs "Total distributed"
- [x] There is a way to paginate / filter the reward history
  - [x] MVP is simple pagination back through epochs
  - [x] Better would be to have the ability to show specific epoch or epochs 
  - [x] There is a default sort on epoch and then by the asset alphabetically
- [x] INclude a view of active rewards 

## Tasks

- [ ] UX - Determine which parts of console you need to surface this
- [ ] UX - Create ACs for each of these
- [ ] Team and stakeholder review
- [ ] Specs reviewed and created or adjusted
- [ ] Implementation
- [ ] Testing (unit and/or e2e)
- [ ] Code review
- [ ] QA review

## Sketch

Figma 

We are going for v2.4 in the Figma

Image

Where there are various levels as indicated in the additional sketch in the Figma

Image

Also using v2.1 of the reward cards. 

https://www.figma.com/file/ywcCZvXdvG28bg0AmCY9mF/Console-v3?type=design&node-id=3166-5839&mode=design&t=GXj0QjK0d268FwQ1-0

### API details

- The total rewards pot comes from the party accounts API with the account type `ACCOUNT_TYPE_VESTED_REWARDS` + `ACCOUNT_TYPE_VESTING_REWARDS`
- The amount available to withdraw is the balance in `ACCOUNT_TYPE_VESTED_REWARDS`
- The vesting base rate is a network param: `rewards.vesting.baseRate`
- Most of the other multipliers can be found on the below API;
  - Vesting multiplier =  `rewardVestingMultiplier`
  - Streak reward multiplier = `rewardDistributionMultiplier`
  - Hoarder reward multiplier = MISSING >> Jeremy is working on this
  - Streak length = `activeFor`
  - Reward hoarder bonus =  `ACCOUNT_TYPE_VESTED_REWARDS` + `ACCOUNT_TYPE_VESTING_REWARDS`
- Activity streak details can be found on network param = `rewards.activityStreak.benefitTiers`
- Reward hoarder bonus tiers can be foudn on network param = `rewards.vesting.benefitTiers`
- The full reward breakdown can be found on the rewards API and can be copied from the Governance implementation and expanded to cover the other reward types. 
 

```
{
  partiesConnection(id:"89fd87b86727fa4d9ab06b9f4e64b51c9b07055e8a0c8f0bc40f0a249e321525"){
    edges{
      node{
        id
        activityStreak{
          activeFor
          isActive
          inactiveFor
          rewardDistributionMultiplier
          rewardVestingMultiplier
          epoch
          tradedVolume
          openVolume
        }
      }
    }
  }
}
```

Note that we have had some issues with pagination on that rewards page on the governance page, which are documented on the ticket below, so we should make sure we handle this nicely on the new console version. 

https://github.com/vegaprotocol/frontend-monorepo/issues/3471

### PR DESCRIPTION
**Pull request by MadalinaRaicu**
# Related issues 🔗

Closes #4532 

# Description ℹ️

- [x] activity streaks
- [x] reward hoarder bonus
- [x] active rewards

# Demo 📺

Screenshot 2023-12-19 at 22 34 29

image

Screenshot 2023-12-19 at 22 53 00

Screenshot 2023-12-19 at 22 34 51

Screenshot 2023-12-19 at 22 35 19

# Technical 👨‍🔧

Details of technical implementation that reviewers may need to be aware of, if applicable.

### CODE REVIEW

File: libs/types/src/global-types-mappings.ts [hunk 1]
```diff
@@ -579,17 +582,31 @@ export const GovernanceTransferKindMapping: GovernanceTransferKindMap = {
...
 };
 
+export const DispatchMetricDescription: DispatchMetricLabel = {
+  DISPATCH_METRIC_LP_FEES_RECEIVED: 'Get rewards for providing liquidity.',
+  DISPATCH_METRIC_MAKER_FEES_PAID:
+    'Get rewards for taking prices on the order book and paying fees.',
+  DISPATCH_METRIC_MAKER_FEES_RECEIVED:
+    'Get rewards for making prices on the order book.',
+  DISPATCH_METRIC_MARKET_VALUE: 'Get rewards for total market value',
+  DISPATCH_METRIC_AVERAGE_POSITION: 'Get rewards for Average position',
```
Comment (candida-d) [inline libs/types/src/global-types-mappings.ts:604]:
```suggestion
  DISPATCH_METRIC_AVERAGE_POSITION: 'Get rewards for having an open position that is consistenly larger than that of other traders.',
```

File: libs/types/src/global-types-mappings.ts [hunk 2]
```diff
@@ -579,17 +582,31 @@ export const GovernanceTransferKindMapping: GovernanceTransferKindMap = {
...
+  DISPATCH_METRIC_RELATIVE_RETURN: 'Get rewards for relative return',
```
Comment (candida-d) [inline libs/types/src/global-types-mappings.ts:605]:
```suggestion
  DISPATCH_METRIC_RELATIVE_RETURN: 'Get rewards for having a high profit in relation to your position size.',
```

File: libs/types/src/global-types-mappings.ts [hunk 3]
```diff
@@ -579,17 +582,31 @@ export const GovernanceTransferKindMapping: GovernanceTransferKindMap = {
...
+  DISPATCH_METRIC_RETURN_VOLATILITY: 'Get rewards for return volatility',
```
Comment (candida-d) [inline libs/types/src/global-types-mappings.ts:606]:
```suggestion
  DISPATCH_METRIC_RETURN_VOLATILITY: 'Get rewards for having the least amount of variance in your returns while you have a position open during the rewards window.',
```

File: libs/types/src/global-types-mappings.ts [hunk 4]
```diff
@@ -579,17 +582,31 @@ export const GovernanceTransferKindMapping: GovernanceTransferKindMap = {
...
+  DISPATCH_METRIC_VALIDATOR_RANKING: 'Get rewards for validator ranking',
```
Comment (candida-d) [inline libs/types/src/global-types-mappings.ts:607]:
```suggestion
  DISPATCH_METRIC_VALIDATOR_RANKING: 'Get rewards if you run a validator node with a high ranking score.',
```

File: libs/types/src/global-types-mappings.ts [hunk 5]
```diff
@@ -579,17 +582,31 @@ export const GovernanceTransferKindMapping: GovernanceTransferKindMap = {
...
   DISPATCH_METRIC_AVERAGE_POSITION: 'Average position',
   DISPATCH_METRIC_RELATIVE_RETURN: 'Relative return',
   DISPATCH_METRIC_RETURN_VOLATILITY: 'Return volatility',
   DISPATCH_METRIC_VALIDATOR_RANKING: 'Validator ranking',
 };
 
+export const DispatchMetricDescription: DispatchMetricLabel = {
+  DISPATCH_METRIC_LP_FEES_RECEIVED: 'Get rewards for providing liquidity.',
+  DISPATCH_METRIC_MAKER_FEES_PAID:
+    'Get rewards for taking prices on the order book and paying fees.',
```
Comment (candida-d) [inline libs/types/src/global-types-mappings.ts:600]:
```suggestion
    'Get rewards for taking prices off the order book and paying fees.',
```

File: libs/types/src/global-types-mappings.ts [hunk 6]
```diff
@@ -579,17 +582,31 @@ export const GovernanceTransferKindMapping: GovernanceTransferKindMap = {
...
+  DISPATCH_METRIC_MAKER_FEES_RECEIVED:
+    'Get rewards for making prices on the order book.',
+  DISPATCH_METRIC_MARKET_VALUE: 'Get rewards for total market value',
```
Comment (candida-d) [inline libs/types/src/global-types-mappings.ts:603]:
```suggestion
  DISPATCH_METRIC_MARKET_VALUE: 'Get rewards if a market you proposed attracts a high trading volume.',
```

**Review by guoguojin — APPROVED** [review]
GQL looks fine

File: apps/trading/components/rewards-container/streaks/reward-hoarder-bonus.tsx [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+
+  const safeProgress = (i: number) => {
+    if (i < userTierIndex) return 100;
+    if (i > userTierIndex) return 0;
+    const progress = vestingDetails.quantumBalance;
+    const total = tiers[i].minimum_quantum_balance;
+    if (new BigNumber(progress).isGreaterThan(total)) return 100;
+    return new BigNumber(progress)
+      .multipliedBy(100)
+      .dividedBy(total)
```
Comment (asiaznik) [inline apps/trading/components/rewards-container/streaks/reward-hoarder-bonus.tsx:59]: The case for infinity and beyond.
**question:** Could this be 0 (zero)?

Comment (MadalinaRaicu) [reply apps/trading/components/rewards-container/streaks/reward-hoarder-bonus.tsx:59]: good catch! i will add an extra check now

**Comment by bwallacee** [comment]
Initial testing: cc @JonRay15 and @MadalinaRaicu 

- **Activity streak is rendered grey for me (I think only when i am at the max tier)**

image


**- I think Vesting should be changed to Vesting Rate**
image

- **It's not very clear what tier I am in for the reward hoarder bonus and acvtivity streak when i am, at the top:**
image
image

- **We are missing the middle breakpoint:**
image

- **Very minor but the dots are centered**
image

- **Epoch appears to be 0 indexed, I'm not sure if this is correct or not**

image

**- Clicking redeem rewards when transfer window is already open doesn't swap the asset over
- Should the from account be pre selected on vested account after using the redeem rewards button**
image
image

- **Should we use better wording than "No row" in the rewards history table:**
image

- **I think the empty activity streak / reward hoarder bonus should be showing still but with me at tier 0?**
image

- **I dont think the reward multiplier is being used correctly. I gained 100 tDAI in a reward while i have a bonus of 3.3x But still only gained 100?**
image
image
image

- **Why don't we see activity streak information in mobile view? In mobile there is currently no way to know what streak you are on and what's next.**

image

- **The scroll bar is white in dark mode instead of dark grey**

**Comment by MadalinaRaicu** [comment]
@bwallacee 

I have pushed a fix for the gray-rendered bar

File: apps/trading/components/rewards-container/Rewards.graphql [hunk 8]
```diff
@@ -36,6 +42,94 @@ query RewardsPage($partyId: ID!) {
...
+          transferId
+          amount
+          epoch
+        }
+      }
+    }
+  }
+}
+
+query ActivityStreak($partyId: ID!) {
```
Comment (mattrussell36) [inline apps/trading/components/rewards-container/Rewards.graphql:113]: We are already fetching the activity streak data in the main RewardsPage query. Why not just use that and save an extra request

File: apps/trading/components/rewards-container/Rewards.graphql [hunk 9]
```diff
@@ -92,3 +186,28 @@ query RewardsEpoch {
     id
   }
 }
+
+query VestingDetails($partyId: ID!) {
+  party(id: $partyId) {
```
Comment (mattrussell36) [inline apps/trading/components/rewards-container/Rewards.graphql:191]: Why not just add to vestingStats in the RewardsPage query

File: apps/trading/components/rewards-container/active-rewards.spec.tsx [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+        transferNode={mockTransferNode}
+        currentEpoch={1}
+        kind={mockRecurringTransfer}
+      />
+    );
+
+    expect(
+      screen.getByText(/Liquidity provision fees received/i)
+    ).toBeInTheDocument();
+    expect(screen.getByText('Entity scope')).toBeInTheDocument();
```
Comment (mattrussell36) [inline apps/trading/components/rewards-container/active-rewards.spec.tsx:107]: Perhaps this might be better to check the correct icon is shown

```
const entityScope = screen.getByText('Entity scope')
expect(within(entityScope.nextElementSibling).getByTestId('icon-team')).toBeInTheDocument();
```

File: apps/trading/components/rewards-container/active-rewards.tsx [hunk 11]
```diff
@@ -0,0 +1,530 @@
...
+                      ? formatNumber(kind.endEpoch - currentEpoch)
+                      : '-',
+                  })}
+                </span>
+              </span>
+            )}
+
+            {
+              <span className="flex flex-col">
+                <span className="text-muted text-xs">{t('Assessed over')}</span>
```
Comment (mattrussell36) [inline apps/trading/components/rewards-container/active-rewards.tsx:352]: This should be a single translation using i18n-next's built in pluralization

File: apps/trading/components/rewards-container/active-rewards.tsx [hunk 12]
```diff
@@ -0,0 +1,530 @@
...
+          {/* TODO use market symbol or market name */}
+          <span>
+            {DispatchMetricLabels[dispatchStrategy.dispatchMetric]}
+            {marketName ? ` • ${marketName}` : ` • ${dispatchAsset?.name}`}
+          </span>
+
+          <div className="flex items-center gap-8 flex-wrap">
+            {kind.endEpoch && (
+              <span className="flex flex-col">
+                <span className="text-muted text-xs">{t('Ends in')}</span>
```
Comment (mattrussell36) [inline apps/trading/components/rewards-container/active-rewards.tsx:339]: This should be a single translation using i18n-next's built in pluralization

File: apps/trading/components/rewards-container/active-rewards.tsx [hunk 13]
```diff
@@ -0,0 +1,530 @@
...
+} from '@vegaprotocol/types';
+import { Card } from '../card/card';
+import { useMemo, useState } from 'react';
+import { useAssetDataProvider } from '@vegaprotocol/assets';
+
+export type Filter = {
+  searchTerm: string;
+};
+
+export const isActiveReward = (node: TransferNode, currentEpoch: number) => {
```
Comment (mattrussell36) [inline apps/trading/components/rewards-container/active-rewards.tsx:43]: nice and readable :-)

File: apps/trading/components/rewards-container/active-rewards.tsx [hunk 14]
```diff
@@ -0,0 +1,530 @@
...
+
+export const applyFilter = (transfer: Transfer, filter: Filter) => {
+  if (
+    transfer.kind.__typename !== 'RecurringTransfer' ||
+    !transfer.kind.dispatchStrategy?.dispatchMetric
+  ) {
+    return false;
+  }
+  if (
+    DispatchMetricLabels[transfer.kind.dispatchStrategy.dispatchMetric]
```
Comment (mattrussell36) [inline apps/trading/components/rewards-container/active-rewards.tsx:73]: (Non blocking): It sounds like @JonRay15 wanted to expand the filter logic. It might be worth using something like https://www.npmjs.com/package/fuse.js if it gets complex

File: apps/trading/components/rewards-container/active-rewards.tsx [hunk 15]
```diff
@@ -0,0 +1,530 @@
...
+              <EntityIcon transfer={transfer} />
+              {entityScope && (
+                <span className="text-muted text-xs">
+                  {EntityScopeLabelMapping[entityScope] || t('Unspecified')}
+                </span>
+              )}
+            </div>
+
+            <div className="flex flex-col gap-2 items-center text-center">
+              <span className="flex flex-col gap-1 text-2xl shrink-1 text-center font-glitch">
```
Comment (mattrussell36) [inline apps/trading/components/rewards-container/active-rewards.tsx:283]: Can we only make the number font-glitch. It doesn't look good when the symbol is fully glitchy

File: apps/trading/components/rewards-container/active-rewards.tsx [hunk 16]
```diff
@@ -0,0 +1,530 @@
...
+            }
+          </div>
+
+          {dispatchStrategy?.dispatchMetric && (
+            <span className="text-muted text-sm h-[2rem]">
+              {t(DispatchMetricDescription[dispatchStrategy?.dispatchMetric])}
+            </span>
+          )}
+
+          <span className="border-[0.5px] border-gray-700" />
```
Comment (mattrussell36) [inline apps/trading/components/rewards-container/active-rewards.tsx:372]: Can you use clight and cdark. Also this can just be border-b for 1px

File: apps/trading/components/rewards-container/active-rewards.tsx [hunk 17]
```diff
@@ -0,0 +1,530 @@
...
+            <div className="flex flex-col gap-2 items-center text-center">
+              <EntityIcon transfer={transfer} />
+              {entityScope && (
+                <span className="text-muted text-xs">
+                  {EntityScopeLabelMapping[entityScope] || t('Unspecified')}
+                </span>
+              )}
+            </div>
+
+            <div className="flex flex-col gap-2 items-center text-center">
```
Comment (mattrussell36) [inline apps/trading/components/rewards-container/active-rewards.tsx:282]: What about a semantic element? The parent of this sections is an h2 so perhaps h3

File: apps/trading/components/rewards-container/rewards-container.tsx [hunk 18]
```diff
@@ -54,6 +61,30 @@ export const RewardsContainer = () => {
 
   const { data: epochData } = useRewardsEpochQuery();
 
+  const { data: partyActivityStreak } = useActivityStreakQuery({
```
Comment (mattrussell36) [inline apps/trading/components/rewards-container/rewards-container.tsx:64]: Can remove these extra queryes if you just add them to the rewards page query

File: apps/trading/components/rewards-container/rewards-container.tsx [hunk 19]
```diff
@@ -114,74 +145,98 @@ export const RewardsContainer = () => {
...
-            const accountsForAsset = rewardAccountsAssetMap[asset.id];
-            const vestedAccount = accountsForAsset?.find(
-              (a) => a.type === AccountType.ACCOUNT_TYPE_VESTED_REWARDS
-            );
-
-            // No vested rewards available to withdraw, so skip over USDT
-            if (!vestedAccount || Number(vestedAccount.balance) <= 0) {
-              return null;
+    <div className="grid auto-rows-min gap-3">
+      <div className="grid auto-rows-min grid-cols-6 gap-3">
```
Comment (mattrussell36) [inline apps/trading/components/rewards-container/rewards-container.tsx:149]: Minor but something about the structure of the grid here is wrong. This grid should only serve as the parent for each gray card

File: apps/trading/components/rewards-container/streaks/activity-streaks.tsx [hunk 20]
```diff
@@ -0,0 +1,218 @@
...
+  streak,
+}: {
+  tiers:
+    | never[]
+    | {
+        minimum_activity_streak: number;
+        reward_multiplier: string;
+        vesting_multiplier: string;
+      }[];
+  streak:
```
Comment (mattrussell36) [inline apps/trading/components/rewards-container/streaks/activity-streaks.tsx:18]: Is it possible to use a generated type here?

File: apps/trading/components/rewards-container/streaks/activity-streaks.tsx [hunk 21]
```diff
@@ -0,0 +1,218 @@
...
+        | undefined
+      )
+    | undefined;
+}) => {
+  const t = useT();
+
+  const getUserTier = () => {
+    if (!streak?.activeFor) return 0;
+    let userTier = 0,
+      i = 0;
```
Comment (mattrussell36) [inline apps/trading/components/rewards-container/streaks/activity-streaks.tsx:40]: Prefer declaring each variable with let explicitly

File: apps/trading/components/rewards-container/streaks/activity-streaks.tsx [hunk 22]
```diff
@@ -0,0 +1,218 @@
...
+          })}
+        </div>
+
+        <div className="flex items-center gap-1">
+          <VegaIcon name={VegaIconNames.STREAK} />
+
+          <span className="flex flex-col">
+            <span>
+              {t('{{epochs}} epochs streak', {
+                epochs: formatNumber(streak?.activeFor || 0),
```
Comment (mattrussell36) [inline apps/trading/components/rewards-container/streaks/activity-streaks.tsx:205]: Might want i18n-next pluralization here?

File: libs/ui-toolkit/src/components/icon/vega-icons/svg-icons/icon-man.tsx [hunk 23]
```diff
@@ -0,0 +1,27 @@
+export const IconMan = ({ size = 14 }: { size: number }) => {
+  return (
+    <svg
```
Comment (mattrussell36) [inline libs/ui-toolkit/src/components/icon/vega-icons/svg-icons/icon-man.tsx:3]: Can you tidy all the svgs

- no xmlns
- no unnecessary fill/stroke attributes
- no hardcoded colors

File: apps/trading/pages/_document.page.tsx [hunk 24]
```diff
@@ -15,7 +15,7 @@ export default function Document() {
           rel="preload"
           href="/AlphaLyrae.woff2"
```
Comment (mattrussell36) [inline apps/trading/pages/_document.page.tsx:16]: This needs changing to just .woff as well

File: apps/trading/components/rewards-container/streaks/reward-hoarder-bonus.tsx [hunk 25]
```diff
@@ -0,0 +1,199 @@
...
+          <div
+            className="grid"
+            style={{
+              gridTemplateColumns:
+                'repeat(' + tiers.length + ', minmax(0, 1fr))',
+            }}
+          >
+            {tiers.map((tier, index) => {
+              return (
+                <div key={index} className="flex justify-end -mr-[2.95rem]">
```
Comment (mattrussell36) [inline apps/trading/components/rewards-container/streaks/reward-hoarder-bonus.tsx:86]: Is there no way to componentize the UI shared between the hoarder bonus and the activity streak?

File: apps/trading/components/rewards-container/streaks/reward-hoarder-bonus.spec.tsx [hunk 26]
```diff
@@ -0,0 +1,59 @@
+import React from 'react';
```
Comment (mattrussell36) [inline apps/trading/components/rewards-container/streaks/reward-hoarder-bonus.spec.tsx:1]: No need to import React

File: apps/trading/components/rewards-container/streaks/activity-streaks.tsx [hunk 27]
```diff
@@ -0,0 +1,218 @@
...
+  const progressBarHeight = 'h-10';
+
+  return (
+    <>
+      <div className="flex flex-col gap-1 w-full">
+        <div className="flex flex-col gap-1">
+          <div
+            className="grid"
+            style={{
+              gridTemplateColumns:
```
Comment (mattrussell36) [inline apps/trading/components/rewards-container/streaks/activity-streaks.tsx:89]: I'm not sure but would flex or auto columns not work here?

File: apps/trading/components/rewards-container/streaks/reward-hoarder-bonus.tsx [hunk 28]
```diff
@@ -0,0 +1,199 @@
...
+    | never[]
+    | {
+        minimum_quantum_balance: string;
+        reward_multiplier: string;
+      }[];
+  vestingDetails?: PartyVestingStats | null;
+}) => {
+  const t = useT();
+
+  const getUserTier = () => {
```
Comment (mattrussell36) [inline apps/trading/components/rewards-container/streaks/reward-hoarder-bonus.tsx:22]: I think this would be better moved into a hook for easier debugging

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: apps/trading/components/rewards-container/Rewards.graphql (hunk 8 above)
Comment (MadalinaRaicu) [reply apps/trading/components/rewards-container/Rewards.graphql:113]: fair enough, updated

File: apps/trading/components/rewards-container/Rewards.graphql (hunk 9 above)
Comment (MadalinaRaicu) [reply apps/trading/components/rewards-container/Rewards.graphql:191]: fair enough, updated

File: apps/trading/components/rewards-container/streaks/reward-hoarder-bonus.spec.tsx (hunk 26 above)
Comment (MadalinaRaicu) [reply apps/trading/components/rewards-container/streaks/reward-hoarder-bonus.spec.tsx:1]: removed

File: apps/trading/pages/_document.page.tsx (hunk 24 above)
Comment (MadalinaRaicu) [reply apps/trading/pages/_document.page.tsx:16]: updated

File: apps/trading/components/rewards-container/active-rewards.tsx (hunk 15 above)
Comment (MadalinaRaicu) [reply apps/trading/components/rewards-container/active-rewards.tsx]: updated

File: apps/trading/components/rewards-container/active-rewards.tsx (hunk 16 above)
Comment (MadalinaRaicu) [reply apps/trading/components/rewards-container/active-rewards.tsx:373]: with border-b and vega clight/cdark it would look more prominent and I was trying to not do that to match the design
Screenshot 2023-12-27 at 18 01 10
Screenshot 2023-12-27 at 18 01 31

File: apps/trading/components/rewards-container/streaks/activity-streaks.tsx (hunk 20 above)
Comment (MadalinaRaicu) [reply apps/trading/components/rewards-container/streaks/activity-streaks.tsx:18]: yes for streaks, for tiers I don't have a generated type as that json is pulled from the network params

File: apps/trading/components/rewards-container/rewards-container.tsx (hunk 18 above)
Comment (MadalinaRaicu) [reply apps/trading/components/rewards-container/rewards-container.tsx]: yes, removed

File: apps/trading/components/rewards-container/active-rewards.tsx (hunk 14 above)
Comment (MadalinaRaicu) [reply apps/trading/components/rewards-container/active-rewards.tsx:73]: I will take a look. https://blog.logrocket.com/fuse-js-dynamic-search-react-app/ 
Once I enrich the transfers with market names and settlement asset the filtering logic would be easier

File: apps/trading/components/rewards-container/streaks/activity-streaks.tsx (hunk 22 above)
Comment (MadalinaRaicu) [reply apps/trading/components/rewards-container/streaks/activity-streaks.tsx:205]: updated

File: apps/trading/components/rewards-container/active-rewards.tsx (hunk 11 above)
Comment (MadalinaRaicu) [reply apps/trading/components/rewards-container/active-rewards.tsx:351]: updated

File: apps/trading/components/rewards-container/active-rewards.tsx (hunk 12 above)
Comment (MadalinaRaicu) [reply apps/trading/components/rewards-container/active-rewards.tsx:340]: updated!

File: apps/trading/components/rewards-container/active-rewards.tsx (hunk 17 above)
Comment (MadalinaRaicu) [reply apps/trading/components/rewards-container/active-rewards.tsx:282]: I only used spans and divs in order to build these parts. But yes, I can do 

image

File: libs/ui-toolkit/src/components/icon/vega-icons/svg-icons/icon-man.tsx (hunk 23 above)
Comment (MadalinaRaicu) [reply libs/ui-toolkit/src/components/icon/vega-icons/svg-icons/icon-man.tsx]: updated

File: apps/trading/components/rewards-container/streaks/reward-hoarder-bonus.tsx (hunk 25 above)
Comment (MadalinaRaicu) [reply apps/trading/components/rewards-container/streaks/reward-hoarder-bonus.tsx:86]: we can, but there are some type differences

File: apps/trading/components/rewards-container/streaks/activity-streaks.tsx (hunk 27 above)
Comment (MadalinaRaicu) [reply apps/trading/components/rewards-container/streaks/activity-streaks.tsx:74]: the streaks could have 12 tiers, after 4 tiers they should have another grid template.

File: apps/trading/components/rewards-container/streaks/reward-hoarder-bonus.tsx (hunk 25 above)
Comment (MadalinaRaicu) [reply apps/trading/components/rewards-container/streaks/reward-hoarder-bonus.tsx:86]: sometimes making them super configurable is not better. I will take a look

File: apps/trading/components/rewards-container/rewards-container.tsx (hunk 19 above)
Comment (MadalinaRaicu) [reply apps/trading/components/rewards-container/rewards-container.tsx:134]: used `flex flex-col w-full gap-3` instead. updated

File: apps/trading/components/rewards-container/streaks/reward-hoarder-bonus.tsx (hunk 25 above)
Comment (MadalinaRaicu) [reply apps/trading/components/rewards-container/streaks/reward-hoarder-bonus.tsx:76]: I made the progress bar fill re-usable. the tiers are very different so the components will be different

File: apps/trading/components/rewards-container/streaks/reward-hoarder-bonus.tsx (hunk 28 above)
Comment (MadalinaRaicu) [reply apps/trading/components/rewards-container/streaks/reward-hoarder-bonus.tsx]: moved in a hook for easier debugging

File: apps/trading/components/rewards-container/streaks/activity-streaks.tsx (hunk 27 above)
Comment (MadalinaRaicu) [reply apps/trading/components/rewards-container/streaks/activity-streaks.tsx:84]: I am not using flex for this 
image

File: apps/trading/components/rewards-container/rewards-history.tsx [hunk 29]
```diff
@@ -258,6 +258,14 @@ export const RewardHistoryTable = ({
     return colDefs;
   }, []);
 
+  if (!pubKey) {
```
Comment (mattrussell36) [inline apps/trading/components/rewards-container/rewards-history.tsx:261]: Why is this added? This table should still be able to render without a connected wallet so you can see all rewards

File: libs/ui-toolkit/src/components/icon/vega-icons/svg-icons/icon-team.tsx [hunk 30]
```diff
@@ -0,0 +1,14 @@
+export const IconTeam = ({ size = 14 }: { size: number }) => {
+  return (
+    <svg width="16" height="16" viewBox="0 0 16 16" fill="none">
+      <g id="Icon">
```
Comment (mattrussell36) [inline libs/ui-toolkit/src/components/icon/vega-icons/svg-icons/icon-team.tsx:4]: No need for this group

File: libs/ui-toolkit/src/components/icon/vega-icons/svg-icons/icon-streak.tsx [hunk 31]
```diff
@@ -0,0 +1,26 @@
+export const IconStreak = ({ size = 14 }: { size: number }) => {
+  return (
+    <svg width="14" height="14" viewBox="0 0 14 14" fill="fillCurrent">
```
Comment (mattrussell36) [inline libs/ui-toolkit/src/components/icon/vega-icons/svg-icons/icon-streak.tsx:3]: All the fill attributes are not required. It will be applied by the fill-current class in vega-icon.tsx

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: apps/trading/components/rewards-container/rewards-history.tsx (hunk 29 above)
Comment (MadalinaRaicu) [reply apps/trading/components/rewards-container/rewards-history.tsx]: updated

File: libs/ui-toolkit/src/components/icon/vega-icons/svg-icons/icon-team.tsx (hunk 30 above)
Comment (MadalinaRaicu) [reply libs/ui-toolkit/src/components/icon/vega-icons/svg-icons/icon-team.tsx]: removed

File: libs/ui-toolkit/src/components/icon/vega-icons/svg-icons/icon-streak.tsx (hunk 31 above)
Comment (MadalinaRaicu) [reply libs/ui-toolkit/src/components/icon/vega-icons/svg-icons/icon-streak.tsx]: removed

File: apps/trading/components/bootstrapper/web3-provider.tsx [hunk 32]
```diff
@@ -24,6 +26,10 @@ export const Web3Provider = ({
   const { ETHEREUM_PROVIDER_URL, ETH_LOCAL_PROVIDER_URL, ETH_WALLET_MNEMONIC } =
     useEnvironment();
 
+  // Query all markets and assets to ensure they are cached
+  useMarketsMapProvider();
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply apps/trading/components/bootstrapper/web3-provider.tsx]: agree. updated!

**Review by mattrussell36 — COMMENTED** [review]
All looks good. Re-reviewing so as not to block this but would be good to get the last changes in

**Review by mattrussell36 — APPROVED** [review]

**Review by bwallacee — APPROVED** [review]
